lib/evm_database.rb

```diff
@@ -183,18 +183,20 @@ def self.configure_rails_handler(monitor)
     rails_handler = ManageIQ::PostgresHaAdmin::RailsConfigHandler.new(:file_path => file_path, :environment => Rails.env)
     _log.info("Configuring database failover for #{file_path}'s #{Rails.env} environment")
 
-    rails_handler.before_failover { LinuxAdmin::Service.new("evmserverd").stop }
-    rails_handler.after_failover do
-      # refresh the rails connection info after the config handler changed database.yml
-      begin
-        ActiveRecord::Base.remove_connection
-      rescue PG::Error
-        # We expect this to fail because it cannot access the database in the cached config
-      end
-      ActiveRecord::Base.establish_connection(Rails.application.config.database_configuration[Rails.env])
+    rails_handler.after_failover do |new_conn_info|
+      # Actions to run upon success
+      unless new_conn_info
```

Comment:
If the failover was successful `new_conn_info` is `nil` ?

Reply:
No, it's false... I've changed this to be more obvious.  Check for the incoming db hash having the `:dbname` key.    If it's not provided, the failover was not successful.

Reply:
Okay I thought maybe I had my unless conditionals inversed one too many times in my head, but looked like this would only run if `new_conn_info` was `nil` if it were a hash then this wouldn't run

Reply:
Yeah, I got rid of the unless because it was confusing me too.  I changed it to:

```ruby
if new_conn_info.try(:key?, :dbname)
  # do the things for a successful failover
end
```

I don't want to rely on the `new_conn_info` being truthy or falsey.  I'd rather assert the new connection information has a field required for connecting to the failover database.